platforms/core-configuration/declarative-dsl-provider/src/main/kotlin/org/gradle/internal/declarativedsl/provider/DeclarativeDslServiceRegistry.kt

```diff
@@ -34,9 +36,16 @@ class DeclarativeDslServiceRegistry : AbstractPluginServiceRegistry() {
 
 internal
 object BuildServices {
+
+    private
+    val schemaRegistry = DefaultDeclarativeSchemaRegistry()
```

Comment:
❌ We should not create this instance here. Instead the `createSchemaRegistry()` function below should create the instance and the `createDeclarativeKotlinScriptEvaluator()` function should take the registry as an injected parameter.

Reply:
Fixed, please approve.